### PR TITLE
Explain how users are backed up and restored

### PR DESCRIPTION
**Pull request by hlcianfagna**
Correction to the user management documentation removing a note that stated users' information could not be backed up and explaining this can be done with snapshots.

### CODE REVIEW

File: docs/admin/user-management.rst [hunk 1]
```diff
@@ -18,7 +18,7 @@ superusers that already exist in the CrateDB cluster. The `CREATE USER`_ and
 When CrateDB is started, the cluster contains one predefined superuser. This
 user is called ``crate``. It is not possible to create any other superusers
 
-Users cannot be backed up or restored.
+Users\' information is backed up together with the cluster\'s metadata when a snapshot is created, and it can be restored independently with the ``USERS`` option of the :ref:`sql-restore-snapshot` command.
```
Comment (hammerhead) [inline docs/admin/user-management.rst:21]: I think a backslash for escaping the apostrophe isn't needed here, it also compiles without.

And you might want to add a line break as per https://github.com/crate/crate/blob/master/devs/docs/write_docs.rst:
> Line length must not exceed 79 characters (except for literals that cannot be wrapped)

File: docs/admin/user-management.rst [hunk 2]
```diff
@@ -18,7 +18,9 @@ superusers that already exist in the CrateDB cluster. The `CREATE USER`_ and
 When CrateDB is started, the cluster contains one predefined superuser. This
 user is called ``crate``. It is not possible to create any other superusers
 
-Users cannot be backed up or restored.
+Users' information is backed up together with the cluster's metadata when a
```
Comment (seut) [inline docs/admin/user-management.rst:21]:
```suggestion
User's information is backed up together with the cluster's metadata when a
```

**Review by seut — COMMENTED** [review]
Thanks for fixing! Spotted a typo.

File: docs/admin/user-management.rst (hunk 2 above)
Comment (hlcianfagna) [reply docs/admin/user-management.rst:21]: Hi Sebastian, it was meant as the information about users plural, but maybe it would be clearer if I reword it to avoid the ' altogether? I could maybe also explain that a restore only creates missing users it does not delete users that did not exist at the time of the backup, what do you think?

File: docs/admin/user-management.rst [hunk 3]
```diff
@@ -18,7 +18,9 @@ superusers that already exist in the CrateDB cluster. The `CREATE USER`_ and
 When CrateDB is started, the cluster contains one predefined superuser. This
 user is called ``crate``. It is not possible to create any other superusers
 
-Users cannot be backed up or restored.
+The definition of all users, including hashes of their passwords, is backed up
+together with the cluster's metadata when a snapshot is created, and it can be
+restored with the ``USERS`` option of the :ref:`sql-restore-snapshot` command.
```
Comment (seut) [inline docs/admin/user-management.rst:23]: It will be also restored when using `RESTORE ALL`, see https://crate.io/docs/crate/reference/en/5.0/sql/statements/restore-snapshot.html. `RESTORE USERS` (or any other section) will restore the users **only**.
```suggestion
together with the cluster's metadata when a snapshot is created, and is 
restored when using the ``ALL`` or ``USERS`` clause of the 
:ref:`sql-restore-snapshot` command.
```

**Review by seut — APPROVED** [review]
👍 thanks!

**Comment by seut** [comment]
please don't forget to squash your commits before merging ;)
